bob-access/src/extractor.rs

```diff
@@ -40,27 +67,32 @@ impl<T: Extractor> ExtractorExt for T {
 
     fn extract_basic(&self) -> Result<Credentials, Error> {
         let mut builder = prepare_builder(self)?;
-        match (self.get_header("username")?, self.get_header("password")?) {
-            (Some(username), Some(password)) => {
-                let creds = builder
-                    .with_username_password(username, password)
-                    .build();
-                Ok(creds)
-            },
-            (None, None) => {
-                if let Some(node_name) = self.get_header("node_name")? {
+        let auth_header = self.get_header("authorization")?;
+        if let Some(auth_header) = auth_header {
+            let mut parts = auth_header.split_whitespace();
+            match (parts.next(), parts.next()) {
+                (Some("Basic"), Some(credentials)) => {
```

Comment:
As I wrote before, the comparison should be case insensitive. You can write simple wrapper struct that support case insensitive comparison with String, so it will be possible to use `match` in the same way

Reply:
Looks like there is a crate, that do exactly what I wrote: https://docs.rs/unicase/latest/unicase/struct.Ascii.html